### ISSUE TITLE
Solid is throwing an error using data from `createResource` as a children of a meta tag from `@solidjs/meta`

### ISSUE THREAD

**Issue by mishimalisa** [solidjs/solid#1195]
### Describe the bug

It seems solid is throwing an error on the server when calling `renderTags()` from `@solidjs/meta`. This happens when the `Title` has a children that depends on the `createResource` from the route data.

Version `1.4.8` throws an error ONLY on the client.
New version `1.5.x` throws an error on the server and client.

Despite the error the `Title` is rendering in the `head` correctly but it does not render the page content using client side navigation.

Minimal example:
```JSX
import { Title } from '@solidjs/meta';

<Show when={data.latest}>
  <Title>{data.latest!.title}</Title>
</Show>
```

Client error:
```
Error: Unrecoverable Hydration Mismatch. No template for key: undefined
```

Server error:
```
TypeError: Cannot read properties of undefined (reading 'async')
```

See reproduction for more details

### Your Example Website or App

https://stackblitz.com/edit/vitejs-vite-ybg9fp?file=src/App.tsx

### Steps to Reproduce the Bug or Issue

1. Run npm run dev and wait for the page to show up
2. Click on the Dashboard link on the page
3. An error is shown that is handled by `ErrorBoundary`.
4. Removing the `ErrorBoundary` shows the stack trace of the error and interestingly the title meta tag is there and correct but page still broken.
5. Now try to access the /dashboard path directly (either by typing or refreshing or clicking on the ssr button) 
6. Notice that server log throws an error

### Expected behavior

Using `@solidjs/meta` and `@solidjs/router` and `createResource` should work together, even if the title is dynamic and came from the `createResource`.

### Platform

- OS: Linux
- Browser: Chrome
- Version: 104.0
- Solid versions I've tested: 1.4.8, 1.5.1, 1.5.2

**Comment by ryansolid** [comment]
Fixed in Solid 1.5.3.

**Comment by mishimalisa** [comment]
Thanks for the amazing work. I just realized that I accidentally used a head tag inside the body that's why it gave a hydration error. But, now the server problem is fixed.
